### PR TITLE
Virtual host using api-dummy: first iteration

### PR DESCRIPTION
**Pull request by giorgiosironi**
- journal's main site is added as a symlink in `sites-enabled`
- an additional virtual host using the `demo` environment is deployed on port 8081, and `/ping` already works
- Gulp is moved as late as possible in the deployment process
- `demo` environment is cleared and placeholder is inserted for additional Gulp commands

All of this is only run in selected environments.

### CODE REVIEW

File: salt/journal/init.sls [hunk 1]
```diff
@@ -1,6 +1,8 @@
 maintenance-mode-start:
     cmd.run:
-        - name: /etc/init.d/nginx stop
+        - name: |
+            rm -f /etc/nginx/sites-enabled/journal.conf
+            /etc/init.d/nginx restart
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: `reload`?

File: salt/journal/init.sls [hunk 2]
```diff
@@ -169,9 +160,35 @@ journal-nginx-vhost:
             - service: nginx-server-service
             - service: php-fpm
 
+journal-nginx-vhost-dummy:
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: Maybe continue to call it `demo`?

File: salt/journal/init.sls [hunk 3]
```diff
@@ -169,9 +160,35 @@ journal-nginx-vhost:
...
+    file.managed:
+        - name: /etc/nginx/sites-enabled/journal-dummy.conf
+        - source: salt://journal/config/etc-nginx-sites-enabled-journal-dummy.conf
+        - template: jinja
+        - require:
+            - nginx-config
+            - nginx-error-pages
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: Not needed?

Comment (giorgiosironi) [inline salt/journal/init.sls]: Hopefully not

File: salt/journal/init.sls [hunk 4]
```diff
@@ -169,9 +160,35 @@ journal-nginx-vhost:
...
+            - journal-node-modules-manual-install
+            - composer-install
+            - journal-nginx-vhost-dummy
+
 maintenance-mode-end:
     cmd.run:
-        - name: /etc/init.d/nginx start
+        - name: |
+            ln -s /etc/nginx/sites-available/journal.conf /etc/nginx/sites-enabled/journal.conf
+            /etc/init.d/nginx restart
```
Comment (thewilkybarkid) [inline salt/journal/init.sls]: Can these just be normal Salt states? So a `file.symlink` with a `watch_in`?

Comment (giorgiosironi) [inline salt/journal/init.sls]: I have always seen watches run at the end of the salt.highstate as fictional added states, but here we need them to be run immediately - we can make them Salt states but not with `watch_in`

Comment (thewilkybarkid) [inline salt/journal/init.sls]: Thought that was the behaviour of `listen_in`. But anyway, guess this is a guarantee.

**Comment by giorgiosironi** [comment]
I have the api-dummy part ready, but would first get this in.

**Comment by thewilkybarkid** [comment]
The `demo` environment will need to be cleared with `bin/console cache:clear --env=test --no-warmup`. This will have to wait for the new CSS to be available, so after `running-gulp` (since we'll need the Gulp critical CSS task to run _after_ this, we'll have to make it a separate task).

File: salt/journal/init.sls (hunk 4 above)
Comment (giorgiosironi) [inline salt/journal/init.sls]: Ah right, that's what I was thinking of - will try it out locally but I suspect the original `service.running` can't be overloaded

**Pull request merged**